### ISSUE TITLE
TheHive release model leads to unexpected regressions and other issues

### ISSUE THREAD

**Issue by Rolinh** [TheHive-Project/TheHive#560]
### Problem Description

First, let me be clear: TheHive is a very nice product to which I did not find any open or closed source equivalent. I am very grateful that you publish this as open source with an open development model.

In order to make TheHive even better, I'd like to bring a topic to the table: I often experience regressions or new bugs when upgrading to what are supposed to be bugfix releases.

The main concern I have is that these "bugfix" releases often are not since they often also include new features (which you sometimes label as "enhancements").
For instance, TheHive [v3.0.7](https://github.com/TheHive-Project/TheHive/milestone/30) introduced the "delete case" feature (#100). This has lead to issues such as #553 or #534. TheHive [v3.0.4](https://github.com/TheHive-Project/TheHive/milestone/27) also brought its share of new features (I count 4 new features) and issues.

Experience now tells me to wait at least 1-2 weeks before upgrading from a release to the next "bugfix" release to make this experience as painless as possible. I believe this could be avoided.
I see that you already included 2 enhancements in the next bugfix release ([3.0.10](https://github.com/TheHive-Project/TheHive/milestone/33))... :cry:
TBH, I'm satisfied with TheHive feature wise and I'd really favor stability over new features.

### Possible Solutions

Change the release model to strictly include bug fixes changes in a bugfix release (ie when only the patch level version number is bumped). At the very least, bump the minor version number if you include new features or "enhancements" in a release.
If you need you need to publish new features more often, maybe release new feature versions more often or switch to a more frequent release model than once or twice a year?

**Comment by nadouani** [comment]
One more point, releases other than hotfix usually introduce new features that require database migrations, which we prefer to minimise and make less frequent as possible.

**Comment by Rolinh** [comment]
> Theoretically, hotfix should not include features, but just bug fixes, but sometimes there are features that could help some people and we don't want to wait for months to ship them.

I definitely understand this point hence my suggestion of switching to a more frequent release models, ie more frequent releases with fewer new features.

**Comment by crackytsi** [comment]
I wonder why there are many issues that are obviously bugs, but they are not planned to be included in the next bugfix Release. 
I really look forward to new features, but open bugs have a higher prioirity for me. E.g. the missing short-reports or the MISP sync errors

**Comment by nadouani** [comment]
This is a valid question. Well it depends on:
- can we reproduce?
- what’s the effort to put in to repeoduce?
- how many users are facing the problem?
- what’s the effort to fix the issue?

And some other questions.

For example, concerning the mini reports issue, I’ve personally spent a lot of time to reproduce the issue, whitout success. After the long discussion we had, we ended saying: we need to create 2 flat analyzers to try to reproduce the issue, which is time consuming without any certainty that we will reproduce the issue, and nobody else asked for it. I’m not saying that the issue does for sure not exist.

To summarize, we really do our very best to fix everything.
We all agree that we don’t want to keep things broken :)

**Comment by crackytsi** [comment]
Thank you very much, for this comments.
Of course you do this, and I have to repeate once more, that your work is so great and I love it so much. 
I will thinking about a way to reduce load from your shoulders but on the other side prevent bugs from getting old ;)

**Comment by petiepooo** [comment]
I'll suggest the LTS model: pick a current stable release with most of the features you want to have out there, declare it a "long-term support" release, and stick to strictly bug fixes on that until it's no longer supported.  After some time, declare a newer LTS version and end updates for the older one.
It adds a little testing load, as you'll always want someone to be able to upgrade from the most current LTS version to the latest nightly version if they need a newly-released feature.  DB migrations accumulate between LTS releases, so regression testing has to include not only nightly-to-nightly but LTS-to-nightly.  As long as LTS-to-nightly always works, when it's time to declare one of the nightly builds as the next LTS release, the DB migration is already working.
Users get their choice between stability (on the LTS track) or features (on the nightly track).
